test/Test/Database/LSMTree/Internal/RawPage.hs

```diff
@@ -254,42 +255,65 @@ prop_entries_all page@(PageLogical xs) bs =
 
 prop_big_insert :: Key -> Maybe BlobRef -> Property
 prop_big_insert k blobref =
-    rawPageLookup rawpage k' === case blobref of
-        Nothing            -> LookupEntryOverflow (Entry.Insert (SerialisedValue $ RB.fromByteString (BS.take size v))) (fromIntegral sfxSize)
-        Just (BlobRef x y) -> LookupEntryOverflow (Entry.InsertWithBlob (SerialisedValue $ RB.fromByteString (BS.take size v)) (BlobSpan x y)) (fromIntegral sfxSize)
+    case rawPageLookup rawpage k' of
+      LookupEntryOverflow (Entry.Insert (SerialisedValue v')) suffixlen
+        | Nothing <- blobref
+        -> v === RB.toByteString v' <>
+                 BS.take (fromIntegral suffixlen) overflowPagesBS
+
+      LookupEntryOverflow (Entry.InsertWithBlob (SerialisedValue v')
+                                                (BlobSpan x' y'))
+                          suffixlen
+        | Just (BlobRef x y) <- blobref
+        -> v === RB.toByteString v' <>
+                 BS.take (fromIntegral suffixlen) overflowPagesBS
+           .&&. (x, y) === (x', y')
+
+      other -> counterexample (show other) False
   where
     page = PageLogical [(k, Insert (Value v), blobref)]
-    (rawpage, sfx) = toRawPage page
+    (rawpage, overflowPages) = toRawPage page
+    overflowPagesBS = RB.toByteString
+                        (mconcat (map rawOverflowPageRawBytes overflowPages))
     k' = SerialisedKey $ RB.fromByteString (unKey k)
 
     -- original value
     v = BS.replicate 5000 42
 
-    size :: Int
-    size = 5000 - sfxSize
-
-    sfxSize :: Int
-    sfxSize = BS.length sfx
-
 prop_single_entry :: Key -> Operation -> Maybe BlobRef -> Property
-prop_single_entry k op blobref = label (show $ BS.null sfx) $
-    rawPageLookup rawpage k' === mkLookupEntry case op of
-        Insert (Value v)       -> case blobref of
-            Nothing            -> Entry.Insert (SerialisedValue $ RB.fromByteString (trim v))
-            Just (BlobRef x y) -> Entry.InsertWithBlob (SerialisedValue $ RB.fromByteString (trim v)) (BlobSpan x y)
-        Mupsert (Value v)      -> Entry.Mupdate (SerialisedValue $ RB.fromByteString (trim v))
-        Delete                 -> Entry.Delete
+prop_single_entry k op blobref = label (show $ null overflowPages) $
+    case rawPageLookup rawpage k' of
+      LookupEntryNotPresent           -> counterexample "LookupEntryNotPresent" False
+      LookupEntry e                   -> compareOpEntry op e 0
+      LookupEntryOverflow e suffixlen -> compareOpEntry op e (fromIntegral suffixlen)
   where
     page = PageLogical [(k, op, blobref)]
-    (rawpage, sfx) = toRawPage page
+    (rawpage, overflowPages) = toRawPage page
+    overflowPagesBS = RB.toByteString
+                        (mconcat (map rawOverflowPageRawBytes overflowPages))
     k' = SerialisedKey $ RB.fromByteString (unKey k)
 
-    trim :: BS.ByteString -> BS.ByteString
-    trim = BS.dropEnd sfxSize
+    compareOpEntry (Insert (Value v))
```

Comment:
add a type signature to compareOpEntry. What it does? Looks complicated, I don't udnerstand